Data race in DMSG test

Log: 
```
==================
WARNING: DATA RACE
Write at 0x00c0021e92f0 by goroutine 48:
  github.com/skycoin/skywire/pkg/dmsg.TestServer_Serve.func5.1()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/dmsg/server_test.go:723 +0x85
Previous write at 0x00c0021e92f0 by goroutine 372:
  github.com/skycoin/skywire/pkg/dmsg.TestServer_Serve.func5()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/dmsg/server_test.go:752 +0x1372
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
Goroutine 48 (running) created at:
  github.com/skycoin/skywire/pkg/dmsg.TestServer_Serve.func5()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/dmsg/server_test.go:722 +0xe28
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
Goroutine 372 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:916 +0x65a
  github.com/skycoin/skywire/pkg/dmsg.TestServer_Serve()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/dmsg/server_test.go:689 +0xab9
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
==================
2019/06/26 09:43:20 SERVE ERR: accept tcp4 127.0.0.1:44271: use of closed network connection
--- FAIL: TestServer_Serve (74.79s)
    --- FAIL: TestServer_Serve/capped_transport_buffer_should_not_result_in_hang (7.59s)
        testing.go:809: race detected during execution of test
    testing.go:809: race detected during execution of test
2019/06/26 09:43:44 127.0.0.1:8081
FAIL
coverage: 80.9% of statements
FAIL    github.com/skycoin/skywire/pkg/dmsg    80.658s
Makefile:63: recipe for target 'test' failed
make: *** [test] Error 1
```

https://travis-ci.org/skycoin/skywire/builds/550250454?utm_source=github_status&utm_medium=notification

--- a/pkg/dmsg/server_test.go
+++ b/pkg/dmsg/server_test.go
@@ -720,7 +720,7 @@ func TestServer_Serve(t *testing.T) {
 
 		// block on `Write`
 		go func() {
-			_, err = aWrTransport.Write(msg)
+			_, err := aWrTransport.Write(msg)
 			require.Error(t, err)
 		}()
 

--- a/pkg/dmsg/server_test.go
+++ b/pkg/dmsg/server_test.go
@@ -719,9 +719,11 @@ func TestServer_Serve(t *testing.T) {
 		}
 
 		// block on `Write`
+		var blockedWriteErr error
+		blockedWriteDone := make(chan struct{})
 		go func() {
-			_, err := aWrTransport.Write(msg)
-			require.Error(t, err)
+			_, blockedWriteErr = aWrTransport.Write(msg)
+			close(blockedWriteDone)
 		}()
 
 		// wait till it's definitely blocked
@@ -748,6 +750,9 @@ func TestServer_Serve(t *testing.T) {
 
 		err = aWrTransport.Close()
 		require.NoError(t, err)
+
+		<-blockedWriteDone
+		require.Error(t, blockedWriteErr)
 
 		err = bWrTransport.Close()
 		require.NoError(t, err)

